frappe/core/doctype/doctype/doctype.json

```diff
@@ -517,12 +524,51 @@
    "fieldname": "email_settings_sb",
    "fieldtype": "Section Break",
    "label": "Email Settings"
+  },
+  {
+   "depends_on": "eval:!doc.istable && !doc.issingle",
+   "fieldname": "sharing_rules_section",
+   "fieldtype": "Section Break",
+   "label": "Sharing Rules"
+  },
+  {
+   "default": "0",
+   "description": "Auto-share the document with the currently assigned user",
+   "fieldname": "auto_share_on_assignment",
+   "fieldtype": "Check",
+   "label": "Auto Share on Assignment"
+  },
+  {
+   "fieldname": "column_break_58",
+   "fieldtype": "Column Break"
+  },
+  {
+   "default": "1",
+   "depends_on": "auto_share_on_assignment",
+   "fieldname": "allow_read",
+   "fieldtype": "Check",
+   "label": "Allow Read",
+   "mandatory_depends_on": "auto_share_on_assignment"
+  },
+  {
+   "default": "0",
+   "depends_on": "auto_share_on_assignment",
+   "fieldname": "allow_write",
```

Comment:
```suggestion
   "fieldname": "allow_write_for_assigned_users",
```
The field name gives the impression that all users will have write permission for the document